dll/win32/shell32/CShellLink.cpp

```diff
@@ -2178,107 +2178,7 @@ HRESULT CShellLink::SetAdvertiseInfo(LPCWSTR str)
-     */
-    static PCWSTR Extensions[] = {L".pif", L".com", L".bat", L".cmd", L".lnk", L".exe", NULL};
-    #define LNK_EXT_INDEX   4   // ".lnk" has index 4 in the array above
-
-    /*
-     * Start at the beginning of the list if PRF_EXECUTABLE is set, otherwise
-     * just use the last element 'NULL' (no extension checking).
-     */
-    i = ((fFlags & PRF_EXECUTABLE) ? 0 : _countof(Extensions) - 1);
-    for (; i < _countof(Extensions); ++i)
-    {
-        /* Ignore shell links ".lnk" if needed */
-        if ((fFlags & PRF_DONTFINDLNK) && (i == LNK_EXT_INDEX))
-            continue;
-
-        Success = (SearchPathW(NULL, pszPath, Extensions[i],
-                               _countof(szPath), szPath, NULL) != 0);
-        if (!Success)
-        {
-            ERR("SearchPathW(pszPath = '%S') failed. Error code: %lu\n", pszPath, GetLastError());
-        }
-        else
-        {
-            ERR("SearchPathW(pszPath = '%S', szPath = '%S') succeeded\n", pszPath, szPath);
-            break;
-        }
-    }
-
-    if (!Success)
-    {
-        ERR("SearchPathW(pszPath = '%S') failed. Error code: %lu\n", pszPath, GetLastError());
-
-        /* We failed, try with PathFindOnPath, as explained by MSDN */
-        // Success = PathFindOnPathW(pszPath, dirs);
-        StringCchCopyW(szPath, _countof(szPath), pszPath);
-        Success = PathFindOnPathW(szPath, dirs);
-        if (!Success)
-        {
-            ERR("PathFindOnPathW(pszPath = '%S') failed\n", pszPath);
-
-            /* We failed again, fall back to building a possible non-existing path */
-            if (!GetFullPathNameW(pszPath, _countof(szPath), szPath, &fname))
-            {
-                ERR("GetFullPathNameW(pszPath = '%S') failed. Error code: %lu\n", pszPath, GetLastError());
-                return FALSE;
-            }
-
-            Success = PathFileExistsW(szPath);
-            if (!Success)
-                ERR("PathFileExistsW(szPath = '%S') failed. Error code: %lu\n", szPath, GetLastError());
-
-            /******************************************************/
-            /* Question: Why this line is needed only for files?? */
-            if (fname && (_wcsicmp(pszPath, fname) == 0))
-                *szPath = L'\0';
-            /******************************************************/
-        }
-        else
-        {
-            ERR("PathFindOnPathW(pszPath = '%S' ==> '%S') succeeded\n", pszPath, szPath);
-        }
-    }
-
-    /* Copy back the results to the caller */
-    StringCchCopyW(pszPath, MAX_PATH, szPath);
-
-    /*
-     * Since the called functions always checked whether the file path existed,
-     * we do not need to redo a final check: we can use instead the cached
-     * result in 'Success'.
-     */
-    return ((fFlags & PRF_VERIFYEXISTS) ? Success : TRUE);
-#endif
-}
+BOOL PathResolveW(LPWSTR path, LPCWSTR *dirs, DWORD flags);
```

Comment:
This may need to be placed in a header.

Reply:
Deleted.
Defined it in `sdk\include\reactos\undocshell.h`.